src/cropduster.js

```diff
@@ -201,33 +203,53 @@ const CD = {
     callback = args.pop();
     options = args[1] || {};
 
+    if (typeof callback !== 'function') {
+      callback = () => { /* do nothing by default */ };
+    }
+
     var msg = "xhr: " + url;
 
-    var req = new XMLHttpRequest();
+    return new Promise(function(resolve, reject) {
```

Comment:
Can we split this method into two — where the behavior of `get` is the same and introduce a new one called `ajax` that returns the promise-only version? (This way we could start deprecating the callback-y form of this). 

The `get` method would end up looking something like:

```js
get(url, options, callback) {
  this.ajax(url, options).then(
    ({ contentType, status, data }) => {
      callback(data, status, contentType);
    }, 
    ({ status }) => {
      callback(null, status);
    }
  ); 
}
```

(Would we want to call it `ajax`? Or, `getPromise`?)

Reply:
I think we'd want to make `get` the primary function--`getCors` uses `get` so you'd end up with `getCorsPromise` or something and it feels a bit weird to have `promise` in function names.  Wouldn't we just be able to do:

```
if(callback) {
  console.log('callbacks are deprecated, prefer promises');
  callback(...);
}
```

Inside the `onerror` / `onsuccess` callbacks?